### PR TITLE
Modify rule S6399: Switch text to progressive education format (APPSEC-425)

### PR DESCRIPTION
**Pull request by gaetan-ferry-sonarsource**

### CODE REVIEW

File: rules/S6399/rationale.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+JSON injections occur when an application builds an XML-formatted string from
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6399/rationale.adoc:1]:
```suggestion
XML injections occur when an application builds an XML-formatted string from
```

File: rules/S6399/impact.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
+=== What is the potential impact?
+
+The consequences of an XML injection attack into an application vary greatly
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6399/impact.adoc:3]:
```suggestion
The consequences of an XML injection attack on an application vary greatly
```

File: rules/S6399/impact.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+later used, the parsing component will raise a technical error. If displayed
+back to the attacker or made available through log files, this technical error
+may disclose sensitive business or technical information.
+
+This scenario, while in general the less severe one, is the most frequently
+encountered. It can combine with any other logic-dependant threat.
+
+==== Internal requests tampering
+
+Some applications communicate with backend micro-services API using XML-based
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6399/impact.adoc:22]:
```suggestion
Some applications communicate with backend micro-services APIs using XML-based
```

File: rules/S6399/common/fix/object.adoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+==== Programmatic object building
+
+In most cases, an application can directly create objects from user input
```
Comment (gaetan-ferry-sonarsource) [inline rules/S6399/common/fix/object.adoc:3]:
```suggestion
In most cases, an application can directly create documents from user input
```

File: rules/S6399/common/fix/introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,6 @@
+In most cases, building XML strings with a direct concatenation of user input
+is discouraged. While not always possible, using type casting or strong
+pattern-based validation can help sanitize tainted inputs.
+
+However, avoiding handling objects' properties as strings by directly
+constructing Java objects should be the preferred way.
```
Comment (egon-okerman-sonarsource) [inline rules/S6399/common/fix/introduction.adoc:6]:
```suggestion
However, directly constructing Java objects should be preferred
over handling the properties of objects as strings.
```
This is easier to read.

**Review by egon-okerman-sonarsource — CHANGES_REQUESTED** [review]
A few changes are necessary, after that should be good to go 👍

File: rules/S6399/java/rule.adoc [hunk 6]
```diff
@@ -1,63 +1,25 @@
...
+
+include::../impact.adoc[]
+
+== How to fix it?
+
+include::./how-to-fix-it/java-se.adoc[]
+
+== Resources
+
+include::../common/resources/docs.adoc[]
```
[the remark this answers is not in the document]
Comment (gaetan-ferry-sonarsource) [reply rules/S6399/java/rule.adoc:13]: Indeed, I left it as a placeholder for later use. Can be safely removed.

**Review by egon-okerman-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**